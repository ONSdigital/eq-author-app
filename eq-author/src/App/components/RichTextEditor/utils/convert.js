import { convertToHTML, convertFromHTML } from "draft-convert";
import { EditorState } from "draft-js";

export const toHTML = entityMap => {
  const entityToHTML = (entity, originalText) => {
    const mapper = entityMap[entity.type];
    return mapper ? mapper(entity) : originalText;
  };

  const convert = convertToHTML({ entityToHTML });

  return editorState => convert(editorState.getCurrentContent());
};

export const fromHTML = htmlToEntity => {
  const convert = convertFromHTML({ htmlToEntity });

  return (html, decorator) =>
    EditorState.createWithContent(convert(html), decorator);
};
