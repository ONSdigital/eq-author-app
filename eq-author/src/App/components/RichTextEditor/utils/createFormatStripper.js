import { mapKeys } from "lodash";
import { filterEditorState } from "draftjs-filters";
import { filterConfig } from "App/components/RichTextEditor/entities/PipedValue";

const mapper = {
  bold: { format: "BOLD", type: "styles" },
  emphasis: { format: "ITALIC", type: "styles" },
  list: { format: "unordered-list-item", type: "blocks" },
  heading: { format: "header-two", type: "blocks" }
};

export default function createFormatStripper(controls) {
  const filterConfiguration = {
    blocks: [],
    styles: [],
    entities: [filterConfig],
    maxNesting: 0,
    whitespacedCharacters: []
  };

  mapKeys(controls, (value, key) => {
    if (mapper[key] && controls[key]) {
      filterConfiguration[mapper[key].type].push(mapper[key].format);
    }
  });

  return editorState => filterEditorState(filterConfiguration, editorState);
}
