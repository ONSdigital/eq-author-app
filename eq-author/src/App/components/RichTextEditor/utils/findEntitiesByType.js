export default entityType => {
  return function findEntitiesByType(contentBlock, callback, contentState) {
    contentBlock.findEntityRanges(character => {
      const entityKey = character.getEntity();
      return (
        character.getEntity() !== null &&
        contentState.getEntity(entityKey).getType() === entityType
      );
    }, callback);
  };
};
