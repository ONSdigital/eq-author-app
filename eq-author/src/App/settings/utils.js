export const getThemeSettingsErrorCount = (questionnaire) =>
  questionnaire?.themeSettings?.validationErrorInfo?.totalCount +
    questionnaire?.validationErrorInfo?.totalCount || 0;
