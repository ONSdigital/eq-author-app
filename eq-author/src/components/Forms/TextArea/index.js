import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { sharedStyles } from "components/Forms/css";
import withChangeHandler from "components/Forms/withChangeHandler";

const StyledTextArea = styled.textarea`
  ${sharedStyles};
  resize: none;
`;

const TextArea = ({ defaultValue, id, rows, onUpdate, ...otherProps }) => (
  <StyledTextArea
    defaultValue={defaultValue}
    rows={rows}
    id={id}
    name={id}
    {...otherProps}
  />
);

TextArea.propTypes = {
  defaultValue: PropTypes.string,
  id: PropTypes.string.isRequired,
  rows: PropTypes.number
};

TextArea.defaultProps = {
  rows: 10
};

export default withChangeHandler(TextArea);
