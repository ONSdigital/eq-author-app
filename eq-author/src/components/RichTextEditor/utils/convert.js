import { convertToHTML, convertFromHTML } from "draft-convert";

export const toHTML = (entityMap) => {
  const entityToHTML = (entity, originalText) => {
    const mapper = entityMap[entity.type];
    return mapper ? mapper(entity) : originalText;
  };

  const convert = convertToHTML({ entityToHTML });

  return (editorState) => convert(editorState.getCurrentContent());
};

export const fromHTML = (nodeToFn) => {
  const htmlToEntity = (nodeName, ...otherArgs) => {
    const entity = Object.entries(nodeToFn)
      .filter(([name]) => name === nodeName)
      .map(([, fn]) => fn(nodeName, ...otherArgs))
      .find((result) => result);

    return entity || null;
  };

  return convertFromHTML({ htmlToEntity });
};
