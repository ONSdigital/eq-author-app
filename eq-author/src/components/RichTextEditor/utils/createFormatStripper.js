import { mapKeys } from "lodash";
import { filterEditorState } from "draftjs-filters";
import { filterConfig as pipedFilterConfig } from "components/RichTextEditor/entities/PipedValue";
import { filterConfig as linkFilterConfig } from "components/RichTextEditor/LinkPlugin";

const mapper = {
  bold: { format: "BOLD", type: "styles" },
  emphasis: { format: "ITALIC", type: "styles" },
  list: { format: "unordered-list-item", type: "blocks" },
  heading: { format: "header-two", type: "blocks" },
};

export default function createFormatStripper(controls) {
  const filterConfiguration = {
    blocks: [],
    styles: [],
    entities: [pipedFilterConfig, linkFilterConfig],
    maxNesting: 0,
    whitespacedCharacters: [],
  };

  mapKeys(controls, (value, key) => {
    if (mapper[key] && controls[key]) {
      filterConfiguration[mapper[key].type].push(mapper[key].format);
    }
  });

  return (editorState) => filterEditorState(filterConfiguration, editorState);
}
