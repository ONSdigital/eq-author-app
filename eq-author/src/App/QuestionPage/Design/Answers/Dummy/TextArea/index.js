import styled from "styled-components";
import PropTypes from "prop-types";
import { sharedStyles } from "components/Forms/css";
import { colors } from "constants/theme";

const DummyTextArea = styled.div`
  ${sharedStyles};
  border-color: ${colors.lightGrey};
  padding: 1.2em 1.2em 1.2em 2em;
  position: relative;
  background-color: transparent;
  width: 50%;
  height: ${props => props.rows + 2}em;
  pointer-events: none;
`;

DummyTextArea.propTypes = {
  rows: PropTypes.number
};

DummyTextArea.defaultPropTypes = {
  rows: 1
};

export default DummyTextArea;
