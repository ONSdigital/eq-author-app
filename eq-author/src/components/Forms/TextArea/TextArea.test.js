import React from "react";
import { render } from "tests/utils/rtl";
import TextArea from "./";

const defaultValue = "I am some text";

describe("components/Forms/TextArea", () => {
  let changeHandler;

  beforeEach(() => {
    changeHandler = jest.fn();
  });

  it("should render correctly", function () {
    expect(
      render(
        <TextArea
          id="text"
          name="text"
          defaultValue={defaultValue}
          onChange={changeHandler}
        />
      ).asFragment()
    ).toMatchSnapshot();
  });

  it("should pass `defaultValue` prop to component when type=text", () => {
    const { getByText } = render(
      <TextArea
        id="text"
        name="text"
        defaultValue={defaultValue}
        onChange={changeHandler}
      />
    );
    expect(getByText(defaultValue)).toBeTruthy();
  });
});
