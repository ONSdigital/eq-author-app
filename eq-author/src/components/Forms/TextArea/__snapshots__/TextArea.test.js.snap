// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Forms/TextArea should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1em;
  border: thin solid #999999;
  padding: 0.5em;
  color: #222222;
  display: block;
  width: 100%;
  -webkit-transition: outline-color 100ms ease-in,border-color 100ms ease-in;
  transition: outline-color 100ms ease-in,border-color 100ms ease-in;
  outline: thin solid transparent;
  resize: none;
}

.c0:hover {
  border-color: #3B7A9E;
  box-shadow: 0 0 0 1pt #3B7A9E;
}

.c0:focus,
.c0:focus-within {
  border-color: transparent;
  outline: 3px solid #FDBD56;
  box-shadow: 0 0 0 3px #FDBD56;
}

.c0::-webkit-input-placeholder {
  color: #a3a3a3;
}

.c0::-moz-placeholder {
  color: #a3a3a3;
}

.c0:-ms-input-placeholder {
  color: #a3a3a3;
}

.c0::placeholder {
  color: #a3a3a3;
}

.c0:focus {
  outline: none;
  border-style: solid;
  border-color: #3B7A9E;
}

.c0[disabled] {
  opacity: 0.8;
  pointer-events: none;
}

<textarea
    class="c0"
    id="text"
    name="text"
    rows="10"
  >
    I am some text
  </textarea>
</DocumentFragment>
`;
